### PR TITLE
Bugfix/fix for glitches in repo

### PR DESCRIPTION
**Pull request by rob-c**
This PR fixes some bugs I've been seeing in private testing. These are all repo handling bugs including https://github.com/ganga-devs/ganga/commit/ddcd3cec49261069cdb426259131cdd0780f614c which was exposed in my private testing.

I would prefer to get these fixes into 6.1.15 if there is time to go to a second pre as (at least) one of these bugs will prevent me from releasing to LHCb.

Apologies that these weren't found sooner but it was only from interactive use that some of these problems were apparent.

### CODE REVIEW

File: python/Ganga/Core/GangaRepository/SubJobXMLList.py [hunk 1]
```diff
@@ -407,10 +413,13 @@ def getAllCachedData(self):
         #logger.debug( "Cache: %s" % str(self._subjobIndexData.keys()) )
         if len(self._subjobIndexData.keys()) == len(self):
             for i in range(len(self)):
```
Comment (milliams) [inline python/Ganga/Core/GangaRepository/SubJobXMLList.py]: This is what `enumerate()` is for.

**Comment by milliams** [comment]
I'll need to check the core tests but at least the ATLAS test are going to need another run because of a bug in `bootstrap.py` so I'm happy to get this included in 6.1.15. It's possible to make PRs against other branches (i.e. directly into `release/6.1.15`) but it really doesn't matter in this case. I'll merge `develop` into the release branch once this is merged and run another set of tests on Monday.

I'm pretty much happy for you to merge when you're ready.

:+1:

File: python/Ganga/Core/GangaRepository/SubJobXMLList.py (hunk 1 above)
Comment (rob-c) [inline python/Ganga/Core/GangaRepository/SubJobXMLList.py]: Probably. But I don't know 100% for certain that all that enumerate calls is len here as I don't want to iterate over the class at all in this step for a few minor reasons. I'll do some investigating in the future but as it stands this PR isn't responsible for this so I'm going to do this elsewhere.

**Comment by rob-c** [comment]
I know it's technically possible to do a PR against any branch but my experience with this has been varied. Merging now.

**Pull request merged**